src/category_theory/instances/groups.lean

```diff
@@ -0,0 +1,63 @@
+/- Copyright (c) 2018 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+
+Introduce Group -- the category of groups.
+
+Currently only the basic setup.
+Copied from monoids.lean.
+-/
+
+import category_theory.concrete_category
+import category_theory.fully_faithful
+
+universes u v
+
+open category_theory
+
+namespace category_theory.instances
+
+/-- The category of groups and group morphisms. -/
+@[reducible] def Group : Type (u+1) := bundled group
+
+instance (G : Group) : group G := G.str
+
+instance concrete_is_group_hom :
+  concrete_category @is_group_hom :=
+⟨by introsI α ia; apply_instance,
+  by introsI α β γ ia ib ic f g hf hg; apply_instance⟩
+
+instance Group_hom_is_group_hom {G₁ G₂ : Group} (f : G₁ ⟶ G₂) :
+  is_group_hom (f : G₁ → G₂) := f.2
+
+instance : has_one Group := ⟨{ α := punit, str := by tidy }⟩
```

Comment:
I think it's better to have a separate `instance : group punit`. In general I think we should try to avoid doing any "work" inside `category_theory` (even if it's trivial like here) that we could instead do outside it without using category theory language.

Rationale: such things might also be useful to people who aren't using `category_theory`, and we don't want the wheel to get reinvented and end up with two wheels.

On the other hand, this one is so trivial that if you don't feel like creating an `algebra.punit_instances`, I don't really mind leaving it as-is.

Reply:
Is this the outstanding issue?

Reply:
I went for the "On the other hand..."

Reply:
I went for [On the one hand]

Reply:
I'm now using the instances that Kenny added.

Reply:
Looks good to me.